cpp/include/cudf/io/functions.hpp

```diff
@@ -421,6 +421,86 @@ void write_parquet(write_parquet_args const& args, rmm::mr::device_memory_resour
+  /// Specify the compression format to use
+  compression_type compression;
+  /// Specify the level of statistics in the output file
+  statistics_freq stats_level;
+  /// Optional associated metadata.
+  const table_metadata_with_nullability *metadata;
+
+  explicit write_parquet_chunked_args(sink_info const& sink_,
+                              const table_metadata_with_nullability *metadata_ = nullptr,
+                              compression_type compression_ = compression_type::AUTO,
+                              statistics_freq stats_lvl_ = statistics_freq::STATISTICS_ROWGROUP)
+      : sink(sink_), metadata(metadata_), compression(compression_), stats_level(stats_lvl_) {}
+};
+
+/**
+ * @brief Forward declaration of anonymous chunked-writer state struct.
+ */
+namespace detail {
+namespace parquet {
+  struct pq_chunked_state;
+};
+};
+
+/**
+ * @brief Begin the process of writing a parquet file in a chunked/stream form.
+ * 
+ * The intent of the write_parquet_chunked_ path is to allow writing of an
+ * arbitrarily large / arbitrary number of rows to a parquet file in multiple passes. 
+ *
+ * The following code snippet demonstrates how to write a single parquet file containing
+ * one logical table by writing a series of individual cudf::tables.
+ * @code
+ *  #include <cudf.h>
+ *  ...
+ *  std::string filepath = "dataset.parquet";
+ *  cudf::experimental::io::write_parquet_chunked_args args{cudf::sink_info(filepath), table->view()};
+ *  ...
+ *  auto state = cudf::write_parquet_chunked_begin(args);
+ *    cudf::write_parquet_chunked(table0, state);
+ *    cudf::write_parquet_chunked(table1, state);
+ *    ...
+ *  cudf_write_parquet_chunked_end(state);
+ * @endcode
+ *
+ * @param[in] args Settings for controlling writing behavior
+ * @param[in] mr Optional resource to use for device memory allocation
+ * 
+ * @returns pointer to an anonymous state structure storing information about the chunked write. this 
+ *          pointer must be passed to all subsequent write_parquet_chunked() and write_parquet_chunked_end()
+ *          calls.
+ */
+std::shared_ptr<detail::parquet::pq_chunked_state> write_parquet_chunked_begin(
+                                                write_parquet_chunked_args const& args, 
+                                                rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource());
+/**
+ * @brief Write a single table as a subtable of a larger logical parquet file/table.
+ * 
+ * All tables passed into multiple calls of this function must contain the same # of columns and have columns
+ * of the same type. 
+ *
+ * @param[in] table The table data to be written.
+ * @param[in] state Opaque state information about the writer process. Must be the same pointer returned
+ *            from write_parquet_chunked_begin()
+ */                                            
+void write_parquet_chunked(table_view const& table, std::shared_ptr<detail::parquet::pq_chunked_state>& state);
```

Comment:
Passing a `shared_ptr` by non-const ref smells fishy. Are you modifying the `shared_ptr`?

Reply:
https://www.modernescpp.com/index.php/c-core-guidelines-passing-smart-pointer

Reply:
Ultimately, write_parquet_chunked_end() does a reset.  The thinking is that the user is just given a widget they ultimately don't need to do anything with except forward it along, so the write_chunked_ code does the cleanup itself.

write_parquet_chunked() could use a const though.

Reply:
Re-opening.  Sorry, my brain wasn't functioning when I read this.  The write_parquet_chunked() case doesn't need a const at all.  I'll update with the changelog conflict/merge update.